src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java

```diff
@@ -71,6 +77,20 @@ public SshSlaveLauncher keyCredentials(String username, String key) {
         cred.username.set(username);
         cred.enterDirectly(key);
         cred.add();
+        credentialUsername = username;
         return this;
     }
+
+    /**
+     * Once a credential has been created for a given slave, this method can be used 
+     * to check whether it has already been rendered in the dropdown.
+     */
+    public void waitForCredentialVisible() {
```

Comment:
Note that when it will not contain the text (but the control would exist) you will never know as the return value of `until` is ignored. I am not sure what are trying to do here (why are you waiting for visual feedback) but you should probably assert it contains the text.